.github/workflows/ztest-short.yml

```diff
@@ -0,0 +1,61 @@
+name: ztest-short
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    env:
+      TEST_DIR: /mnt/ztest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install --yes -qq build-essential autoconf libtool gdb lcov \
+          git alien fakeroot wget curl bc fio acl \
+          sysstat mdadm lsscsi parted gdebi attr dbench watchdog ksh \
+          nfs-kernel-server samba rng-tools xz-utils \
+          zlib1g-dev uuid-dev libblkid-dev libselinux-dev \
+          xfslibs-dev libattr1-dev libacl1-dev libudev-dev libdevmapper-dev \
+          libssl-dev libffi-dev libaio-dev libelf-dev libmount-dev \
+          libpam0g-dev pamtester python-dev python-setuptools python-cffi \
+          python3 python3-dev python3-setuptools python3-cffi
+    - name: Autogen.sh
+      run: |
+        sh autogen.sh
+    - name: Configure
+      run: |
+        ./configure --enable-debug --enable-debuginfo
+    - name: Make
+      run: |
+        make --no-print-directory -s pkg-utils pkg-kmod
+    - name: Install
+      run: |
+        sudo dpkg -i *.deb
+        # Update order of directories to search for modules, otherwise
+        #   Ubuntu will load kernel-shipped ones.
+        sudo sed -i.bak 's/updates/extra updates/' /etc/depmod.d/ubuntu.conf
+        sudo depmod
+        sudo modprobe zfs
+    - name: Tests
+      run: |
+        sudo mkdir -p $TEST_DIR
+        # run for 20 minutes to have a total runner time of 30 minutes
+        sudo /usr/share/zfs/zloop.sh -t 1200 -l -m3 -f $TEST_DIR -c $TEST_DIR/core
+    - name: Prepare artifacts
+      if: failure()
+      run: |
+        sudo dmesg > $TEST_DIR/core/dmesg
+        sudo cp /var/log/syslog $TEST_DIR/core/
+        sudo chmod +r -R $TEST_DIR/core/
```

Comment:
@behlendorf I didn't get failures, so i think that `zloop.sh -c $TEST_DIR/core` should place wanted artifacts in $TEST_DIR/core, am I right here?

Reply:
Right.  Everything under `$TEST_DIR/core` would be the useful artifacts.   Though we probably want to exclude the actual pool which is stored in the `vdev` directory simply due to its size and the actual core file.

To make it easier to iterate on testing the log gathering you might make a change like this to `ztest` so it immediately fails.

```diff
diff --git a/cmd/ztest/ztest.c b/cmd/ztest/ztest.c
index 1c4da20..077ee5c 100644
--- a/cmd/ztest/ztest.c
+++ b/cmd/ztest/ztest.c
@@ -7733,6 +7733,8 @@ main(int argc, char **argv)
        char *fd_data_str = getenv("ZTEST_FD_DATA");
        struct sigaction action;
 
+       ASSERT(0);
+
        (void) setvbuf(stdout, NULL, _IOLBF, 0);
 
        dprintf_setup(&argc, argv);
```